iroh-util/src/lock.rs

```diff
@@ -1,79 +1,304 @@
-use anyhow::Result;
-use file_guard::{FileGuard, Lock};
+use anyhow::{anyhow, Result as AnyhowResult};
 use std::fs::File;
+use std::io::prelude::*;
+use std::io::ErrorKind;
+use std::io::Write;
 use std::path::PathBuf;
-use std::rc::Rc;
+use sysinfo::{Pid, ProcessExt, ProcessStatus, System, SystemExt};
+use thiserror::Error;
+use tracing::warn;
 
-/// Manages a lock file used to track if an iroh program
-/// is already running.
-/// The lock is released either when the object is dropped
-/// or when the program stops.
+pub fn acquire_or_exit(lock: &mut ProgramLock) -> &mut ProgramLock {
+    match lock.is_locked() {
+        Ok(false) => {
+            if let Err(e) = lock.acquire() {
+                eprintln!("error locking {}: {}", lock.program_name(), e);
+                std::process::exit(crate::exitcodes::ERROR);
```

Comment:
please import things, and don't use the full paths, eg `process::exit(exitcodes::Error)` would be the main convention here